src/Native/Unix/System.Native/pal_interfaceaddresses.cpp

```diff
@@ -57,6 +57,10 @@ extern "C" int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onI
         }
         
         assert(result == actualName);
+        if (current->ifa_addr == nullptr)
```

Comment:
Could/should this check be moved to the beginning of the loop rather than here, so we don't do any work for such cases?